docs/dense-retrieval.md

```diff
@@ -167,6 +162,45 @@ map                     all     0.3706
 recall_1000             all     0.9736
 ```
 
-To evaluate with on-the-fly query encoding, replace `--encoded-queries` with our pretrained encoder model
-on [Hugging Face](https://huggingface.co/castorini/tct_colbert-msmarco/tree/main) 
+To evaluate with on-the-fly query encoding with our pretrained encoder model
+on [Hugging Face](https://huggingface.co/castorini/tct_colbert-msmarco/tree/main) add
 `--encoder castorini/tct_colbert-msmarco`. The encoding will run on CPU by default. To enable GPU, add `--device cuda:0`.
+
+## DPR Retrieval
+
+Vladimir Karpukhin, Barlas Oğuz, Sewon Min, Patrick Lewis, Ledell Wu, Sergey Edunov, Danqi Chen, Wen-tau Yih, [Dense Passage Retrieval for Open-Domain Question Answering](https://arxiv.org/abs/2004.04906), Preprint 2020.
+
+
+Download question&answers from Natural Question dev set
+```bash
+$ wget https://dl.fbaipublicfiles.com/dpr/data/retriever/nq-dev.qa.csv
+```
+
+Convert the `nq-dev.qa.csv` to topics file in `json` format
+```bash
+$ python scripts/dpr/convert_qas_csv_to_topic_json.json --input nq-dev.qa.csv --output nq-dev.qa.json
```

Comment:
Just a note for future reference, maybe this should be added on the anserini end? Okay like this for now.

Reply:
yeah

Reply:
Maybe this is a task for @KaiSun314 ?